docs/chia-blockchain/architecture/mempool.md

```diff
@@ -109,6 +127,62 @@ spend bundle also has one aggregate signature, which is a combination of every s
 
 For performance reasons, the chia-blockchain codebase currently creates only smaller blocks (less than 50% of the maximum size) in order to keep the blockchain smaller and easier to run. This "throttle" is likely to be removed in future versions, after additional optimizations have been performed.
 
+### Enhanced Block Creation (Chia 2.5.5+)
+
+Starting in Chia 2.5.5, a new, more efficient block creation algorithm is available that can be enabled via configuration. This algorithm provides improved performance and better resource utilization during block creation.
+
+**Configuration**: Set `full_node:block_creation` to `1` in your config file to enable the new algorithm.
+
+**Benefits**:
+
+- Improved block creation performance
+- Better memory management during block creation
+- Enhanced handling of high-transaction-volume scenarios
+- More efficient resource utilization
+
+**Configurable Timeout**: Block creation now supports a configurable timeout setting via `full_node:block_creation_timeout`, allowing node operators to fine-tune the process based on their network conditions and requirements.
+
+**Example Configuration**:
+
+```yaml
+full_node:
+  block_creation: 1 # Enable new algorithm
+  block_creation_timeout: 30 # 30 second timeout
+```
+
 ## Updating the Mempool
 
 After a new block is added to the blockchain, all full nodes must look at the coins that were spent in that new block, and remove them from the mempool. The full node does not need to reapply every transaction again, since Chia coin spends are deterministic and sandboxed (see the [Coin Set Intro page](/chia-blockchain/coin-set-model/intro) for more information). The full node only needs to look at the spent coins in the new block, and if there are any transactions that spend one of those coins, they are removed from the mempool. This means the mempool can be very large, the codebase can be simple, and high performance can be achieved.
+
+## Schema Changes and Migration (Chia 2.5.5+)
+
+Starting in Chia 2.5.5, the mempool has undergone backwards incompatible schema changes to support the new fast-forward functionality and optimizations.
+
+### Backwards Incompatible Changes
+
+**Available in**: Chia 2.5.5 and later versions
+
+The mempool schema has been updated to support:
+
+- **Fast-forward processing** for singleton and vault transactions
+- **Enhanced transaction deduplication** mechanisms
+- **Improved memory management** and performance optimizations
+
+### Migration Notes
+
+**Important**: These changes are backwards incompatible. Nodes running versions prior to 2.5.5 may experience issues when connecting to nodes running 2.5.5+.
+
+**Required Actions**:
+
+- **Upgrade all nodes** to Chia 2.5.5+ simultaneously
+- **Clear mempool data** if upgrading from versions prior to 2.5.5
```

Comment:
I think you should add a note that this happens automatically when you restart the farmer. Otherwise, people will wonder how to do it.